src/htm/algorithms/SpatialPooler.cpp

```diff
@@ -172,7 +172,7 @@ void SpatialPooler::setDutyCyclePeriod(UInt dutyCyclePeriod) {
 Real SpatialPooler::getBoostStrength() const { return boostStrength_; }
 
 void SpatialPooler::setBoostStrength(Real boostStrength) {
-  NTA_CHECK(boostStrength >= 0.0f);
+  NTA_CHECK(boostStrength == 0.0f or boostStrength >= 1.0) << "Boost strength must be >= 1.0, or exactly 0.0 (=disabled).";
```

Comment:
Why is this a restriction?  Does it break with `0 < boostStrength < 1` ?

Reply:
I can remove this, my idea was that if we boost, we want to do so in a positive way, not "silencing" the column. 
Actually, I'd like to move to "your" log-boost function that is parameterless.

Reply:
reverted these limits. It could make sense to use 0 < strength < 1, to "soften" the boosting.